target/linux/generic/hack-5.15/790-SFP-GE-T-ignore-TX_FAULT.patch

```diff
@@ -1,47 +1,19 @@
+diff --git a/drivers/net/phy/sfp.c b/drivers/net/phy/sfp.c
+index 1df5d5008ab2..d47ce11e0ae1 100644
 --- a/drivers/net/phy/sfp.c
 +++ b/drivers/net/phy/sfp.c
-@@ -1803,6 +1803,7 @@ static int sfp_sm_mod_probe(struct sfp *
- 	struct sfp_eeprom_id id;
- 	bool cotsworks_sfbg;
- 	bool cotsworks;
-+	bool oem_ge_t;
- 	u8 check;
- 	int ret;
- 
-@@ -1851,6 +1852,10 @@ static int sfp_sm_mod_probe(struct sfp *
- 		}
- 	}
- 
-+	/* Some cheap SFP-GE-T modules always indicate TX fault */
-+	oem_ge_t = !memcmp(id.base.vendor_name, "OEM             ", 16) &&
-+		   !memcmp(id.base.vendor_pn, "SFP-GE-T    ", 12);
-+
- 	/* Cotsworks do not seem to update the checksums when they
- 	 * do the final programming with the final module part number,
- 	 * serial number and date code.
-@@ -1946,8 +1951,8 @@ static int sfp_sm_mod_probe(struct sfp *
- 	else
- 		sfp->module_t_start_up = T_START_UP;
- 
--	if (!memcmp(id.base.vendor_name, "HUAWEI          ", 16) &&
--	    !memcmp(id.base.vendor_pn, "MA5671A         ", 16))
-+	if ((!memcmp(id.base.vendor_name, "HUAWEI          ", 16) &&
-+	    !memcmp(id.base.vendor_pn, "MA5671A         ", 16)) || oem_ge_t)
- 		sfp->tx_fault_ignore = true;
- 	else
- 		sfp->tx_fault_ignore = false;
-@@ -2404,10 +2409,12 @@ static void sfp_check_state(struct sfp *
- 	mutex_lock(&sfp->st_mutex);
- 	state = sfp_get_state(sfp);
- 	changed = state ^ sfp->state;
--	if (sfp->tx_fault_ignore)
-+	if (sfp->tx_fault_ignore) {
- 		changed &= SFP_F_PRESENT | SFP_F_LOS;
--	else
-+		state &= ~SFP_F_TX_FAULT;
-+	} else {
- 		changed &= SFP_F_PRESENT | SFP_F_LOS | SFP_F_TX_FAULT;
-+	}
- 
- 	for (i = 0; i < GPIO_MAX; i++)
- 		if (changed & BIT(i))
+@@ -369,6 +369,14 @@ static const struct sfp_quirk sfp_quirks[] = {
+ 		.modes = sfp_quirk_2500basex,
+ 		.fixup = sfp_fixup_ignore_tx_fault,
+ 	}, {
++		// OEM SFP-GE-T is Huawei MA5671A compatible,
++		// It can operate at 2500base-X, but report 1.2GBd
++		// NRZ in their EEPROM
++		.vendor = "OEM",
++		.part = "SFP-GE-T",
++		.modes = sfp_quirk_2500basex,
++		.fixup = sfp_fixup_ignore_tx_fault,
```

Comment:
Unfortunately this fixup is not enough and/or doesn't get applied due to the quirky name (note spaces vs. 0-bytes in both product and vendor name...).
```
[   17.635421] sfp sfp1: module transmit fault indicated
[   23.035427] sfp sfp1: module persistently indicates fault, disabling
```

Reply:
Exactly, these two rows we had for HALny HL-GSFP. Maybe, it requires the identical/similar workaround than just ignoring TX_fault?

Reply:
Yes, that's what I tried above (removing `sfp_quirk_2500basex` line) and still doesn't work (see above).

In my original patch I had to:
 * use custom string matching as vendor and product name are padded with spaces, but not to 16 chars total, but in case of the product name only to 12 chars total.
ethtools -m shows `Vendor PN                                 : SFP-GE-T    ____`
 * make sure TX_FAULT line is really ignored always, that's why I also had to patch what you are then removing here: https://github.com/openwrt/openwrt/pull/10593/commits/084eda2c41fc50124e6fb7add9f7c6e4dacc3c17#diff-5d3dd57341f75047a5027c7e24a74a3caee97a7cfb81dd4f050be75d04dd89a0L41